src/pricer/pricer.ts

```diff
@@ -0,0 +1,89 @@
+import { BigNumber as BigNumberJs } from 'bignumber.js';
+import log from 'loglevel';
+import {
+  ExchangeApiName,
+  INTERMEDIATE_CURRENCY,
+  tokenToApi,
+  createExchangeApi,
+} from './utils';
+
+const getExchangeRate = async (
+  sourceCurrency: string,
+  targetCurrency: string,
+  intermediateCurrency = INTERMEDIATE_CURRENCY
+) => {
+  log.debug('RelayPricer: intermediary Currency', intermediateCurrency);
+
+  const sourceExchangeRate = await queryExchangeRate(
+    sourceCurrency,
+    intermediateCurrency
+  );
+
+  const targetExchangeRate = await queryExchangeRate(
+    targetCurrency,
+    intermediateCurrency
+  );
+
+  if (sourceExchangeRate.isEqualTo(0) || targetExchangeRate.isEqualTo(0)) {
+    const message = `Currency conversion for pair ${sourceCurrency}:${targetCurrency} not found in current exchange api`;
```

Comment:
this kinda duplicates the error in each api class. I know that the check is different, but I'd say it should be handled in a single place only.

Reply:
well I just checked and the errors are different, are similar but describe different issues on different points, and I think that cannot be merged or centralized.